packages/scripts/src/multi-chain/utils.js

```diff
@@ -0,0 +1,109 @@
+const { ZERO_ADDRESS } = require("@uma/common");
+const { getAbi, getAddress } = require("@uma/contracts-node");
+
+const fetch = (...args) => import("node-fetch").then(({ default: fetch }) => fetch(...args));
+
+async function fetchFullL1Whitelist(l1Web3, l1ChainId) {
+  const l1TokenWhitelist = new l1Web3.eth.Contract(
+    getAbi("AddressWhitelist"),
+    await getAddress("AddressWhitelist", l1ChainId)
+  );
+
+  const l1AddedToWhitelistTokens = (
+    await l1TokenWhitelist.getPastEvents("AddedToWhitelist", { fromBlock: 0, toBlock: "latest" })
+  ).map((event) => event.returnValues.addedAddress);
+
+  const l1RemovedFromWhitelistTokens = (
+    await l1TokenWhitelist.getPastEvents("RemovedFromWhitelist", { fromBlock: 0, toBlock: "latest" })
+  ).map((event) => event.returnValues.removedAddress);
+
+  const whitelistedAddressArray = l1AddedToWhitelistTokens.filter((address) => {
+    return !l1RemovedFromWhitelistTokens.includes(address);
+  });
+
+  const l1Store = new l1Web3.eth.Contract(getAbi("Store"), await getAddress("Store", l1ChainId));
+
+  const finalFeesArray = await Promise.all(
+    whitelistedAddressArray.map((address) => l1Store.methods.finalFees(address).call())
+  );
+
+  const symbols = (
+    await Promise.allSettled(
+      whitelistedAddressArray.map((address) =>
+        new l1Web3.eth.Contract(getAbi("ERC20"), address).methods.symbol().call()
+      )
+    )
+  ).map((result) => {
+    return result.status === "fulfilled" ? result.value : "NO-SYMBOL";
+  });
+
+  return whitelistedAddressArray.map((l1TokenAddress, index) => {
+    return { l1TokenAddress, finalFee: finalFeesArray[index], symbol: symbols[index] };
+  });
+}
+
+async function findL2TokenForL1Token(l2Web3, l2ChainId, l1TokenAddress) {
+  if (l2ChainId == 10) {
+    const foundOnChain = await _findL2TokenForOvmChain(l2Web3, l1TokenAddress);
+    if (foundOnChain != ZERO_ADDRESS) return foundOnChain;
+    else return await _findL2TokenFromTokenList(l2ChainId, l1TokenAddress);
+  }
+  if (l2ChainId == 288) {
+    return await _findL2TokenForOvmChain(l2Web3, l1TokenAddress);
+  }
+
+  if (l2ChainId == 42161) {
+    return await _findL2TokenFromTokenList(l2ChainId, l1TokenAddress);
+  }
+}
+
+async function _findL2TokenFromTokenList(l2ChainId, l1TokenAddress) {
+  if (l2ChainId == 10) {
+    const response = await fetch("https://static.optimism.io/optimism.tokenlist.json");
```

Comment:
if l2ChainId !== 10 or 42161 where will it error?

Reply:
if the chain id is neither of these then it returns a blank array from the token lists. for example boba's list is built up exclusively from the `_findL2TokenForOvmChain` method, not using any lists, as their is no token list like this for boba.